Mathlib/GameTheory/Auction/Secondprice.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2024 Wang Haocheng. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ma Jiajun, Wang Haocheng
+-/
+import Mathlib.Data.Fintype.Basic
+import Mathlib.Data.Finset.Lattice
+import Mathlib.Data.Real.Basic
+
+/-!
+# Auction Theory
+
+This file formalizes core concepts and results in auction theory. It includes the definitions of
+first-price and second-price auctions, as well as several fundamental results and helping lemmas.
+
+## Main Definitions
+
+- `Auction`: Defines an auction with bidders and their valuations.
+- `maxBid`: Function that computes the highest bid given a bidding function.
+- `winner`: Identifies the winner of the auction as the bidder with the highest bid.
+- `maxBidExcluding`: Function that computes the highest bid excluding a given participant.
+- `dominant`: Defines a dominant strategy in the context of a auction.
+- `Secondprice`: Computes the second highest bid in the auction.
```

Comment:
I have changed the file name to `Basic.lean` to store the current elements of auction theory in one place. Rather than add a new file contains all the basic definitions in `Basic.lean` other than `SecondPrice.lean` is because that this content does not yet encompass numerous theorems. 

This way, all related theorems can reside in `Basic.lean` for future modifications and additions.

Reply:
It is okay to have few theorems. The idea is, if I want to work on auctions, I shouldn't have to include second price auction stuff unless I need it.